Modules/_collectionsmodule.c

```diff
@@ -1050,39 +1071,52 @@ deque_index(dequeobject *deque, PyObject *const *args, Py_ssize_t nargs)
         start = stop;
     assert(0 <= start && start <= stop && stop <= Py_SIZE(deque));
 
-    for (i=0 ; i < start - BLOCKLEN ; i += BLOCKLEN) {
-        b = b->rightlink;
-    }
-    for ( ; i < start ; i++) {
-        index++;
-        if (index == BLOCKLEN) {
-            b = b->rightlink;
-            index = 0;
-        }
-    }
+    index = start;
+    b = deque_get_block(deque, &index);
 
-    n = stop - i;
+    n = stop - start ;
     while (--n >= 0) {
         CHECK_NOT_END(b);
         item = b->data[index];
-        cmp = PyObject_RichCompareBool(item, v, Py_EQ);
+        cmp = PyObject_RichCompareBool(item, value, Py_EQ);
         if (cmp > 0)
-            return PyLong_FromSsize_t(stop - n - 1);
+            return stop - n - 1;
         if (cmp < 0)
-            return NULL;
+            return -1;
         if (start_state != deque->state) {
```

Comment:
Let's remove the mutation tracking and testing entirely.  Instead, just check that the found *index* is within the current deque size.  If it isn't, just return *Py_None* and skip the deletion step.  This will bring the code back in-sync with what *list.remove()* does.

Reply:
Just to double-check: you want to remove the check for imitation from `deque_index`, from `deque_remove` or from both?